### DISCUSSION TITLE
Add generics to type `data` and `newData` in DataTableRowEditSaveEvent, DataTableRowEditCancelEvent and DataTableRowEditInitEvent interfaces. DataTable component

### DISCUSSION THREAD

**Discussion by lucasdavidferrero** [primefaces/community#3002] [PrimeVue]
The current definition is:
```ts
export interface DataTableRowEditSaveEvent {
  originalEvent: Event  
  data: any
  newData: any
  field: string
  index: number
}
```
An example handling `@row-edit-save` in `DataTable` component with current typescript definitions:
```ts
async function onRowEditSave (event: DataTableRowEditSaveEvent) {
  let { newData, data, index } = event;
  const newName = newData.fullname // We don't get autocomplete here. It's also unsafe if doing comparations
  const oldName = data.fullname
  if (newName !== oldName) {
    rowData.value[index] = newData  // assigning is unsafe here. rowData entry's type should match newData's type.
  }
}
```

I think **adding a generic for typing `data` and `newData`** properties will improve DX:
```ts
export interface DataTableRowEditSaveEvent<TData> {
  originalEvent: Event  
  data: TData
  newData: TData
  field: string
  index: number
}
```

This situation of having `data` and `newData` happens in other data structures related with the `DataTable` component such as: `DataTableCellEditCompleteEvent`, `DataTableCellEditCancelEvent`, `DataTableCellEditInitEvent`, `DataTableRowCollapseEvent`, `DataTableRowExpandEvent` among others...

**Comment by j0rgedev** [comment]
Hey @lucasdavidferrero. I think you could try opening a PR! I support this

**Comment by lucasdavidferrero** [comment]
This will be solved in version 4.3.0
